File: task3/task.cpp
#include <iostream>
#include <numeric>

using namespace std;

long long power(long long a, long long b)
{
    long long result = 1;

    for (long long i = 0; i < b; i++)
        result *= a;

    return result;
}

int main()
{
    int a, b;
    cin >> a >> b;

    // Если ряд расходится
    if (b == 1)
    {
        cout << "infinity";
        return 0;
    }

    // Формулы для первых степеней
    long long numerator = 0;
    long long denominator = 1;

    if (a == 1)
    {
        numerator = b;
        denominator = (b - 1) * (b - 1);
    }
    else if (a == 2)
    {
        numerator = b * (b + 1);
        denominator = power(b - 1, 3);
    }
    else if (a == 3)
    {
        numerator = b * (b * b + 4 * b + 1);
        denominator = power(b - 1, 4);
    }

    long long g = gcd(numerator, denominator);

    numerator /= g;
    denominator /= g;

    cout << numerator << "/" << denominator;

    return 0;
}

File: task3/task.go
package main

import (
	"fmt"
)

// Функция возведения в степень
func power(a, b int64) int64 {

	result := int64(1)

	for i := int64(0); i < b; i++ {
		result *= a
	}

	return result
}

// Функция нахождения НОД
func gcd(a, b int64) int64 {

	for b != 0 {
		a, b = b, a%b
	}

	return a
}

func main() {

	var a, b int64

	// Ввод параметров
	fmt.Print("Введите a и b: ")
	fmt.Scan(&a, &b)

	// Если b = 1, ряд расходится
	if b == 1 {
		fmt.Print("infinity\n\n")
		return
	}

	var numerator int64
	var denominator int64

	// Формула для a = 1
	if a == 1 {

		numerator = b
		denominator = (b - 1) * (b - 1)

	} else if a == 2 {

		// Формула для a = 2
		numerator = b * (b + 1)
		denominator = power(b-1, 3)

	} else if a == 3 {

		// Формула для a = 3
		numerator = b * (b*b + 4*b + 1)
		denominator = power(b-1, 4)
	}

	// Нахождение НОД
	g := gcd(numerator, denominator)

	// Сокращение дроби
	numerator /= g
	denominator /= g

	// Вывод результата
	fmt.Printf("%d/%d\n\n", numerator, denominator)
}
